### PR TITLE
Bump lib to next-gen.

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Bump lib to next-gen.
- Fix code to work again.

### CODE REVIEW

**Comment by juancarlospaco** [comment]
@joliveros Please, if you can when you can make a Review, thanks...

File: routes/index.js [hunk 1]
```diff
@@ -36,20 +38,23 @@ module.exports = function(app) {
...
-            console.log(err);
-            if (err) {
-                response.status(500).send(err);
+            url: url,
+        };
+
+        // Place an outbound call to the user, using the TwiML instructions
+        // from the /outbound route
+        client.api.v2010.accounts(config.accountSid).calls.create(options)
+          .then((error, message) => {
```
Comment (hortega) [inline routes/index.js]: The `then` callback does not seem right. You can either pass to `then` 2 callbacks: the first for a fulfilled promise and in this case with a single argument. The second for a rejected promise also with a single Error argument. 
The other option is to pass only a `onFulfilled` callback to `then` and chain afterwards a `catch` method with a single `onReject` callback.

Look at an example of the first option in the Twilio-node specs https://github.com/twilio/twilio-node/blob/next-gen/spec/integration/rest/api/v2010/account/call.spec.js#L29

And of the second in https://github.com/TwilioDevEd/server-notifications-node/blob/next-gen/twilioClient.js

**Comment by hortega** [comment]
Apart from an issue with the client promise callback, the rest looks good to me. Good work!

File: package.json [hunk 2]
```diff
@@ -17,16 +17,16 @@
...
   "engines" : { "node" : ">=4.1.0 <5.5.0" },
   "dependencies": {
-    "body-parser": "^1.6.5",
-    "express": "^4.8.5",
-    "twilio": "^1.6.0",
-    "morgan": "^1.2.3",
-    "jade": "1.7.x"
+    "body-parser": "^1.15.2",
+    "express": "^4.14.0",
+    "twilio": ">=3.0.0-rc.13",
```
Comment (joliveros) [inline package.json]: static version as per Agustin

**Comment by joliveros** [comment]
Everything looks good @juancarlospaco. I would follow Hector's advice and go through with those changes regarding Promises. 

Nice work Sir!

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+sudo: false
+
+language: node_js
+
+node_js:
+  - 'stable'
+  - '4.4'
+
+install: npm install
```
Comment (acamino) [inline .travis.yml]: `install` and `script` steps are redundant.
- The language `node_js` has a [default test script](https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Default-Test-Script).
- The same applies to how the dependencies are installed. https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Travis-CI-uses-npm

Could you please remove them?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+script: npm test
+
+env:
+  global:
+    - NODE_TLS_REJECT_UNAUTHORIZED=0
+    - TWILIO_ACCOUNT_SID="AC3f899f99b9ec999999ded99c09c99e09"  # fake
+    - TWILIO_AUTH_TOKEN="89b09c9bff999999a9c9999c9eda9e91"  # fake
+    - TWILIO_NUMBER="+5411511111381"  # fake
+
+cache:
```
Comment (acamino) [inline .travis.yml]: There is no need to use `cache` in this project. Probably some C&P issue but why to have `pip` over here. 😛

File: README.md [hunk 5]
```diff
@@ -5,6 +5,8 @@ step-by-step tutorial, [visit this link](https://twilio.com/docs/howto/click-to-
 
 [Read the full tutorial here](https://www.twilio.com/docs/tutorials/walkthrough/click-to-call/node/express)!
 
+[![Build Status](https://travis-ci.org/TwilioDevEd/clicktocall-node.svg?branch=master)](https://travis-ci.org/TwilioDevEd/clicktocall-node)
```
Comment (acamino) [inline README.md]: Could you please take a look at the [README Express template](https://github.com/twilio/docs-guidelines/tree/fb01ef506476a9d9694cddc22721bd0c7a88c11e/templates/README/node/express)? We'd like to have some sort of standardisation across all the tutorials.

File: package.json [hunk 6]
```diff
@@ -17,16 +17,16 @@
...
   "engines" : { "node" : ">=4.1.0 <5.5.0" },
   "dependencies": {
-    "body-parser": "^1.6.5",
-    "express": "^4.8.5",
-    "twilio": "^1.6.0",
-    "morgan": "^1.2.3",
-    "jade": "1.7.x"
+    "body-parser": "^1.15.2",
+    "express": "^4.14.0",
+    "twilio": "3.0.0-rc.13",
```
Comment (acamino) [inline package.json]: 💯 for using a strict version.

The tutorials are an extension of Twilio's documentation -- https://www.twilio.com/docs/libraries/node#installation-nextgen.

File: routes/index.js [hunk 7]
```diff
@@ -2,11 +2,13 @@ var path = require('path');
...
 var bodyParser = require('body-parser');
-var twilio = require('twilio');
-var config = require("../config");
+const Twilio = require('twilio');
+const config = require('../config');
+
 
 // Create a Twilio REST API client for authenticated requests to Twilio
-var client = twilio(config.accountSid, config.authToken);
+const client = new Twilio(config.accountSid, config.authToken);
```
Comment (acamino) [inline routes/index.js]: To address this one please refer to https://github.com/TwilioDevEd/marketing-notifications-node/pull/6/files#r87578365

File: routes/index.js [hunk 8]
```diff
@@ -36,23 +38,26 @@ module.exports = function(app) {
...
-                    message: 'Thank you! We will be calling you shortly.'
-                });
-            }
-        });
+            url: url,
+        };
+
+        // Place an outbound call to the user, using the TwiML instructions
+        // from the /outbound route
+        client.api.v2010.accounts(config.accountSid).calls.create(options)
```
Comment (acamino) [inline routes/index.js]: Not sure why you're using this interface? Why not using something the canonical approach https://www.twilio.com/docs/api/rest/making-calls?

**Comment by juancarlospaco** [comment]
Sorry, the _"Fix code to work again"_ gets inherited on the title from the first commit and I didnt notice, good catch there, I fixed the title...

**Pull request closed without merging**